rtos/TARGET_CORTEX/rtx5/RTX/Source/rtx_lib.c

```diff
@@ -693,21 +693,15 @@ __WEAK int _mutex_initialize(mutex *m) {
 }
 
 // Acquire mutex
-#if !defined(__ARMCC_VERSION) || __ARMCC_VERSION < 6010050
-__USED
-#endif
-__WEAK void _mutex_acquire(mutex *m);
+__WEAK __USED void _mutex_acquire(mutex *m);
```

Comment:
used was added to fix this issue https://github.com/ARM-software/CMSIS_5/issues/129. Looking at this, do we still need weak? (the PR referenced at the bottom there, was closed, and `RTX_NO_MULTITHREAD_CLIB ` was mentioned there).

Reply:
__WEAK and __USED seems not to make sense if used both. As discussed earlier this function cannot be defined __WEAK because there is already a weak version provided in the C library. Hence the linker cannot distinguish which weak version to use if no non-weak version is given. If its really needed for mbed to get rid of the internal implementation that is part of RTX we need to find a completely different solution.